### ISSUE TITLE
x/text p.Sprint outpust full key

### ISSUE THREAD

**Issue by mvrhov** [golang/go#26519]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
1.11b2

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?
linux/amd64

### What did you do?
```
T.Sprint(message.Key("login.form.label.password", "Password"))
```
### What did you expect to see?
Password

### What did you see instead?
{login.form.label.password Password}

`T.Sprintf(message.Key("login.form.label.password", "Password"))` returns the expected output, however it's use is not necessary in many such cases, where there is no parameters for the string even the lint tool is showing warnings in such cases.

**Comment by meirf** [comment]
@mvrhov, can you please elaborate a bit on this:

> ... however it's use is not necessary in many such cases, where there is no parameters for the string even the lint tool is showing warnings in such cases.

Is it possible you're not using the print* functions as intended? Note the difference in behavior in the following code:

```
package main

import (
	"fmt"
	"golang.org/x/text/language"
	"golang.org/x/text/message"
)

type foo struct {
	x float64
}

func main() {
	message.SetString(language.German, "archive(noun)", "archive")

	p := message.NewPrinter(language.German)

	p.Print(123456.78)
	fmt.Println()
	p.Print(foo{123456.78})
	fmt.Println()
	p.Print("archive(noun)")
	fmt.Println()
	p.Printf("archive(noun)")
	fmt.Println()

	fmt.Println(p.Sprint(message.Key("login.form.label.password", "Password")))
	fmt.Println(p.Sprintf(message.Key("login.form.label.password", "Password")))
}
```
Result:
```
123.456,78
{123.456,78}
archive(noun)
archive
{login.form.label.password Password}
Password
```
Note the differences: Print/Sprint vs Printf/Sprintf.
Print/Sprint: Accept `a ...interface{}`. Just do localized formatting. 
Printf/Sprintf: Accept `key Reference, a ...interface{}`. Does translation and fallback.

So for your code:
`T.Sprint(message.Key("login.form.label.password", "Password"))`
we expect that it will just do localized formatting of each field which in this case is just the contents of the Reference struct. Sprint doesn't view the input as a Reference to translate+fallback - just an arbitrary object to format.

Please let me know if I'm mistaken.

**Comment by mvrhov** [comment]
Yeah. I've seen that the function signature is different I'd say that it's still a bit unexpected. 
And that's the reason I'd like to keep this open if you don't mind.

**Comment by meirf** [comment]
> And that's the reason I'd like to keep this open if you don't mind.

Okay, but can you please be more specific on what you are proposing. What API change/addition would you like to happen or what documentation would you like to change or what bug should be fixed?

> ... however it's use is not necessary in many such cases, where there is no parameters for the string even the lint tool is showing warnings in such cases.

This is still a bit vague to me. Can you please help me understand by elaborating more?

**Comment by mvrhov** [comment]
>Okay, but can you please be more specific on what you are proposing. What API change/addition would you..

The API could figure out itself if based on the first parameter which behavior it should use. e.g if the message with fallback is passed then it uses one path otherwise there is no fallback it uses the 2nd path. Unless this really slows things down.

For the 2nd part. Please disregard it. The reason for the 2nd part is because I've been using different languages for the past 20years+ and have thought that the behavior is as following.. Functions ending with `f` are used only when you have to insert/format the passed parameters e.g `fmt.Sprintf("This string has a couple of %s parameters, %d", "string", 10)` where the functions not ending with `f` are to be used when you don't have such case e.g `fmt.Sprint("This string has no parameters")`.

**Comment by mikegleasonjr** [comment]
I too was confused by all of this and lost quite a bit of time trying to understand why `Sprint` would not localize my key while `Sprintf` would.

It would be great if each argument in `Print/Sprint` would "Do a translation and fallback.". At least that's what I propose.
